transformer_engine/paddle/csrc/common.h

```diff
@@ -0,0 +1,146 @@
+/*************************************************************************
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See LICENSE for license information.
+ ************************************************************************/
+#pragma once
+
+#include <cublasLt.h>
+#include <transformer_engine/cast.h>
+#include <transformer_engine/gemm.h>
+#include <transformer_engine/logging.h>
+#include <transformer_engine/transformer_engine.h>
+#include <vector>
+
+#include "paddle/extension.h"
+
+namespace transformer_engine {
+namespace paddle_ext {
+// Each tensor here is shape (N, ) holding all scaling
+// data for a single FP8 block, e.g. LayerNormLinear
+class FP8TensorMeta {
+ public:
+    paddle::Tensor scale;
+    paddle::Tensor scale_inv;
+    paddle::Tensor amax_history;
+};
+
+// Used as named indices on the `scale`, `scale_inv`,
+// and `amax` tensors in the `FP8TensorMeta` class.
+enum FP8FwdTensors {
```

Comment:
Enum classes are more manageable:
```suggestion
enum class FP8FwdTensors {
```
A similar change should be made to [`FP8BwdTensors`](https://github.com/NVIDIA/TransformerEngine/blob/e2b847007d5bed6877b2ca8f8856539fa42b84d4/transformer_engine/paddle/csrc/common.h#L41).

Reply:
Done